test/mpi/init/session.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpi.h"
+#include <stdio.h>
+#include <assert.h>
+#include "mpitest.h"
+
+/* This is example 10.8 from MPI Standard 4.0 */
+
+int errs = 0;
+
+int library_foo_test(void);
+void library_foo_init(void);
+void library_foo_finalize(void);
+
+int main(int argc, char *argv[])
+{
+#ifdef MULT_INIT
+    int rank, size, provided;
+    int num_repeat = 1;
+
+    if (argc > 1) {
+        num_repeat = atoi(argv[1]);
+        /* basic sanity check */
+        assert(num_repeat > 0 && num_repeat < 100);
+    }
+
+    MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    for (int i = 0; i < num_repeat; i++) {
+        library_foo_test();
+    }
+    MPI_Finalize();
+#else
+    int rank = library_foo_test();
+#endif
+    if (rank == 0 && errs == 0) {
+        printf("No Errors\n");
+    }
+    return MTestReturnValue(errs);
+}
+
+static MPI_Session lib_shandle = MPI_SESSION_NULL;
+static MPI_Comm lib_comm = MPI_COMM_NULL;
+
+int library_foo_test(void)
+{
+    int rank, size;
+
+    library_foo_init();
+
+    MPI_Comm_size(lib_comm, &size);
+    MPI_Comm_rank(lib_comm, &rank);
+
+    int sum;
```

Comment:
> sum is not initialized. It may be set to a garbage value if MPI_Reduce does not work correctly.
> It is a very small effort to make the test more robust (i.e., initialize sum) :-)

The correct `MPI_Reduce` should work correctly. I don't understand the point of making the test more "robust" -- I mean, if the test fails, it will indicate there is a bug in our implementation, isn't that a good thing?